packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -712,6 +730,106 @@ Map<String, String> parseAdbDeviceProperties(String str) {
+///
+/// Example output:
+///
+/// Applications Memory Usage (in Kilobytes):
+/// Uptime: 441088659 Realtime: 521464097
+///
+/// ** MEMINFO in pid 16141 [io.flutter.demo.gallery] **
+///                    Pss  Private  Private  SwapPss     Heap     Heap     Heap
+///                  Total    Dirty    Clean    Dirty     Size    Alloc     Free
+///                 ------   ------   ------   ------   ------   ------   ------
+///   Native Heap     8648     8620        0       16    20480    12403     8076
+///   Dalvik Heap      547      424       40       18     2628     1092     1536
+///  Dalvik Other      464      464        0        0
+///         Stack      496      496        0        0
+///        Ashmem        2        0        0        0
+///       Gfx dev      212      204        0        0
+///     Other dev       48        0       48        0
+///      .so mmap    10770      708     9372       25
+///     .apk mmap      240        0        0        0
+///     .ttf mmap       35        0       32        0
+///     .dex mmap     2205        4     1172        0
+///     .oat mmap       64        0        0        0
+///     .art mmap     4228     3848       24        2
+///    Other mmap    20713        4    20704        0
+///     GL mtrack     2380     2380        0        0
+///       Unknown    43971    43968        0        1
+///         TOTAL    95085    61120    31392       62    23108    13495     9612
+///
+///  App Summary
+///                        Pss(KB)
+///                         ------
+///            Java Heap:     4296
+///          Native Heap:     8620
+///                 Code:    11288
+///                Stack:      496
+///             Graphics:     2584
+///        Private Other:    65228
+///               System:     2573
+///
+///                TOTAL:    95085       TOTAL SWAP PSS:       62
+///
+///  Objects
+///                Views:        9         ViewRootImpl:        1
+///          AppContexts:        3           Activities:        1
+///              Assets:        4        AssetManagers:        3
+///        Local Binders:       10        Proxy Binders:       18
+///        Parcel memory:        6         Parcel count:       24
+///     Death Recipients:        0      OpenSSL Sockets:        0
+///             WebViews:        0
+///
+///  SQL
+///          MEMORY_USED:        0
+///   PAGECACHE_OVERFLOW:        0          MALLOC_SIZE:        0
+/// ...
+///
+/// For more information, see https://developer.android.com/studio/command-line/dumpsys.
+@visibleForTesting
+AndroidMemoryInfo parseMeminfoDump(String input) {
+  Iterable<String> lines = input.split('\n');
+  lines = lines
+    .skipWhile((String line) => !line.contains('App Summary'))
+    .takeWhile((String line) => !line.contains('TOTAL'));
+  final AndroidMemoryInfo androidMemoryInfo = AndroidMemoryInfo();
+  for (String line in lines) {
+    if (!line.contains(':')) {
```

Comment:
To avoid mixing styles:

Maybe move to after the `takeWhile`, like `.where((String line) => line.contains(':'))`

Then you can `.forEach((String line) {...})` instead of a for loop.